scripts/setup.sh

```diff
@@ -85,12 +85,11 @@ function run() {
     echo "leak:v8::internal" >> ${SUPPRESSION_FILE}
     echo "leak:node::CreateEnvironment" >> ${SUPPRESSION_FILE}
     echo "leak:node::Init" >> ${SUPPRESSION_FILE}
-    echo "export ASAN_SYMBOLIZER_PATH=${llvm_toolchain_dir}/bin/llvm-symbolizer" >> ${config}
-    echo "export MSAN_SYMBOLIZER_PATH=${llvm_toolchain_dir}/bin/llvm-symbolizer" >> ${config}
+    echo "export ASAN_SYMBOLIZER_PATH=$(which llvm-symbolizer)" >> ${config}
     echo "export UBSAN_OPTIONS=print_stacktrace=1" >> ${config}
     echo "export LSAN_OPTIONS=suppressions=${SUPPRESSION_FILE}" >> ${config}
-    echo "export ASAN_OPTIONS=detect_leaks=1:symbolize=1:abort_on_error=1:detect_container_overflow=1:check_initialization_order=1:detect_stack_use_after_return=1" >> ${config}
-    echo 'export MASON_SANITIZE="-fsanitize=address,undefined,integer,leak -fno-sanitize=vptr,function"' >> ${config}
+    echo "export ASAN_OPTIONS=symbolize=1:abort_on_error=1:detect_container_overflow=1:check_initialization_order=1:detect_stack_use_after_return=1" >> ${config}
+    echo 'export MASON_SANITIZE="-fsanitize=address,undefined -fno-sanitize=vptr,function"' >> ${config}
```

Comment:
Why disable Leak checking? Last I checked `leak` was default enabled in linux but not OS X, so this is needed to be able to check leaks on osx.

Reply:
Sure, I'll re-enable them. This got mixed up with my attempts to fix `asan` builds on travis

Reply:
OK, reverting back is causing travis to fail - https://travis-ci.com/github/mapnik/node-mapnik/jobs/457466380 
I don't quite understand what's the issue here, yet. Let me know if you have some ideas, cheers

Reply:
I notice two problems:

```
/home/travis/build/mapnik/node-mapnik/.mason/mason.sh: line 559:  7865 Aborted                 (core dumped) $(which fakeroot) tar xzf "${MASON_BINARIES_PATH}"
Please run setup.sh first
Please run setup.sh first
scripts/postgis/setup_and_run.sh: line 4: mason-postgis-config.env: No such file or directory
The command "source scripts/postgis/setup_and_run.sh" failed and exited with 1 during .
```

^^ looks like it might be unrelated to ASAN stuff, and something needing fixing overall that regressed. Can you ticket or look into that?

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==7903==ERROR: AddressSanitizer: SEGV on unknown address (pc 0x7f18b28d9edc bp 0x7f18b1d2fe32 sp 0x7ffcae9c0180 T0)
==7903==The signal is caused by a READ memory access.
==7903==Hint: this fault was caused by a dereference of a high value address (see register values below).  Dissassemble the provided pc to learn which register was used.
```
I've never seen anything like this before. So I'm unsure if it is a problem with the node-mapnik code or with the sanitizer.

Reply:
maybe try removing the `-fno-sanitize-recover=all` in the .travis.yml?

Reply:
I'll try ^ but I'm pretty sure the issue is only indirectly related to asan stuff, thanks

Reply:
As expected removing `-fno-sanitize-recover=all1 didn't reveal anything useful https://travis-ci.com/github/mapnik/node-mapnik/jobs/457532596
What's strange is that the failure is happening here https://github.com/mapnik/node-mapnik/blob/master/scripts/postgis/setup.sh#L10-L12
Looks like either : 
 
`GDAL_DATA_VALUE=$(mason prefix libgdal ${GDAL_VERSION})/share/gdal/` 

or

`mason install postgis ${POSTGIS_VERSION}` causing a  segfault, weird. 


```* Already installed at /home/travis/build/mapnik/node-mapnik/mason_packages/linux-x86_64/libgdal/2.1.3
* Downloading binary package https://mason-binaries.s3.amazonaws.com/linux-x86_64/postgis/2.3.2-1.tar.gz
/home/travis/build/mapnik/node-mapnik/.mason/mason.sh: line 559:  7869 Aborted                 (core dumped) $(which fakeroot) tar xzf "${MASON_BINARIES_PATH}"
Please run setup.sh first
Please run setup.sh first
scripts/postgis/setup_and_run.sh: line 4: mason-postgis-config.env: No such file or directory```

Reply:
Hmm.. in .travis I see

```
  # TODO: re-enable detect_leaks=0 once we can build and test against an asan sanitized libc++
        - export ASAN_OPTIONS=fast_unwind_on_malloc=0:detect_leaks=0:${ASAN_OPTIONS}
        - source scripts/postgis/setup_and_run.sh
```
Which implies that `detect_leaks=1` was disabled ^^^ 

https://github.com/mapnik/node-mapnik/blob/master/.travis.yml#L70-L72

Reply:
https://github.com/mapnik/node-mapnik/commit/cd5f6b419ca447840b67cf26c10e050de260b91b 
@springmeyer

Reply:
To make `detect_leaks` working we need sanitized deps e.g libc++ also. Those checks have been disabled for quite a while https://github.com/mapnik/node-mapnik/commit/cd5f6b419ca447840b67cf26c10e050de260b91b